install/kubernetes/helm/subcharts/mixer/templates/deployment.yaml

```diff
@@ -28,6 +28,7 @@
           - --monitoringPort={{ .Values.global.monitoringPort }}
           - --address
           - unix:///sock/mixer.socket
+          - --numCheckCacheEntries=0
```

Comment:
should this be made the default value for the arg ?

Reply:
https://github.com/istio/istio/blob/df0d817c5ecf6f3fab94cb599ddd90c363c4d533/mixer/pkg/server/args.go#L127

Reply:
I'm not it's a good idea to _change_ defaults. If someone customized the mixer deployment file already, then it's an extra (and not as obvious) thing to miss. Explicit flag is very obvious.
What do you think?

Reply:
Generally, I'd agree with you. But if the check cache is *fundamentally* broken, I don't see how we could, in good conscience, leave it enabled by default.

@geeknoid @mandarjog thoughts here?

Reply:
I found one cause of the problem. It's caused by the change of the underlying type for the stringmap values here:

https://github.com/istio/istio/blob/master/mixer/pkg/checkcache/keyShape.go#L198-200

At some point it got swapped to `attribute.StringMap` and the case statement did not have a default clause. 

I think the lesson here is that we have insufficient test coverage for the check cache. I am not convinced this is the only mistake in the cache.